### ISSUE TITLE
Help Wanted: Completing the API Reference on Website

### ISSUE THREAD

**Issue by fabian-hiller** [fabian-hiller/valibot#287]
> [!IMPORTANT]
> The following description is outdated. All synchronous APIs are already fully documented. We are just missing a few asyncronous APIs. If you still want to help, feel free to comment below.

This issue is used to coordinate the expansion of our API reference. The first reference was created by @kazizi55 in #220. Please use the already added references as a template. You can also use this issue if you encounter problems or have questions.

If you are interested in helping us extend the API reference, please add a comment to this issue with the API you want to take over. I will summarize the current state in the tables below.

<details>
  <summary>Schemas</summary><br />

| API | Author | Pull Request | Merged |
| -- | -- | -- | -- |
| any | @fabian-hiller | - | ✅ |
| array | @kazizi55 | #330 | ✅ |
| bigint | @fabian-hiller | - | ✅ |
| blob | @fabian-hiller | - | ✅ |
| boolean | @wout-junius | #316 | ✅ |
| date | @wout-junius | #315 | ✅ |
| enum\_ | @fabian-hiller | - | ✅ |
| instance | @fabian-hiller | - | ✅ |
| intersect | @fabian-hiller | - | ✅ |
| literal | @fabian-hiller | - | ✅ |
| map | @fabian-hiller | - | ✅ |
| nan | @fabian-hiller | - | ✅ |
| never | @fabian-hiller | - | ✅ |
| nonNullable | @fabian-hiller | - | ✅ |
| nonNullish | @fabian-hiller | - | ✅ |
| nonOptional | @fabian-hiller | - | ✅ |
| null\_ | @fabian-hiller | - | ✅ |
| nullable | @sqmasep | #328 | ✅ |
| nullish | @sqmasep | #329 | ✅ |
| number | @kazizi55 | #331 | ✅ |
| object | @kazizi55 | #357 | ✅ |
| optional | @sqmasep | #321 | ✅ |
| picklist | @sqmasep | #312 | ✅ |
| record | @fabian-hiller | - | ✅ |
| recursive | @fabian-hiller | - | ✅ |
| set | @fabian-hiller | - | ✅ |
| special | @fabian-hiller | - | ✅ |
| string | @kazizi55 | #220 | ✅ |
| symbol | @wout-junius | #317 | ✅ |
| tuple | @fabian-hiller | - | ✅ |
| undefined\_ | @fabian-hiller | - | ✅ |
| union | @fabian-hiller | - | ✅ |
| unknown | @mwskwong | #365 | ✅ |
| variant | @fabian-hiller | - | ✅ |
| void\_ | @fabian-hiller | - | ✅ |

</details>

<details>
  <summary>Methods</summary><br />

| API | Author | Pull Request | Merged |
| -- | -- | -- | -- |
| brand | @fabian-hiller | - | ✅ |
| coerce | @fabian-hiller | - | ✅ |
| fallback | @sqmasep | #337 | ✅ |
| forward | @fabian-hiller | - | ✅ |
| getDefault | @fabian-hiller | - | ✅ |
| getDefaults | @fabian-hiller | - | ✅ |
| getFallback | @fabian-hiller | - | ✅ |
| getFallbacks | @fabian-hiller | - | ✅ |
| is | @fabian-hiller | - | ✅ |
| keyof | @fabian-hiller | - | ✅ |
| merge | @fabian-hiller | - | ✅ |
| omit | @fabian-hiller | - | ✅ |
| parse | @fabian-hiller | - | ✅ |
| partial | @fabian-hiller | - | ✅ |
| pick | @fabian-hiller | - | ✅ |
| required | @fabian-hiller | - | ✅ |
| safeParse | @fabian-hiller | - | ✅ |
| transform | @fabian-hiller | - | ✅ |
| unwrap | @fabian-hiller | - | ✅ |

</details>

<details>
  <summary>Validations</summary><br />

| API | Author | Pull Request | Merged |
| -- | -- | -- | -- |
| bic | @ariskemper | #386 | ✅ |
| bytes | @fabian-hiller | - | ✅ |
| creditCard | @ariskemper | #388 | ✅ |
| cuid2 | @fabian-hiller | - | ✅ |
| custom | @fabian-hiller | - | ✅ |
| decimal | @fabian-hiller | - | ✅ |
| email | @fabian-hiller | - | ✅ |
| emoji | @fabian-hiller | - | ✅ |
| endsWith | @fabian-hiller | - | ✅ |
| every | @fabian-hiller | - | ✅ |
| excludes | @fabian-hiller | - | ✅ |
| finite | @fabian-hiller | - | ✅ |
| hash | @ariskemper | #392 | ✅ |
| hexadecimal | @ariskemper | #414 | ✅ |
| hexColor | @ariskemper | #401 | ✅ |
| iban | @ariskemper | | |
| imei | @fabian-hiller | - | ✅ |
| includes | @fabian-hiller | - | ✅ |
| integer | @fabian-hiller | - | ✅ |
| ip | @fabian-hiller | - | ✅ |
| ipv4 | @fabian-hiller | - | ✅ |
| ipv6 | @fabian-hiller | - | ✅ |
| isoDate | @fabian-hiller | - | ✅ |
| isoDateTime | @fabian-hiller | - | ✅ |
| isoTime | @fabian-hiller | - | ✅ |
| isoTimeSecond | @fabian-hiller | - | ✅ |
| isoTimestamp | @fabian-hiller | - | ✅ |
| isoWeek | @fabian-hiller | - | ✅ |
| length | @fabian-hiller | - | ✅ |
| mac | @ariskemper | #407 | ✅ |
| mac48 | @ariskemper | #407 | ✅ |
| mac64 | @ariskemper | #407 | ✅ |
| maxBytes | @fabian-hiller | - | ✅ |
| maxLength | @fabian-hiller | | |
| maxSize | @fabian-hiller | - | ✅ |
| maxValue | @fabian-hiller | - | ✅ |
| mimeType | @fabian-hiller | - | ✅ |
| minBytes | @fabian-hiller | - | ✅ |
| minLength | @fabian-hiller | | |
| minSize | @fabian-hiller | - | ✅ |
| minValue | @fabian-hiller | - | ✅ |
| multipleOf | @fabian-hiller | - | ✅ |
| notBytes | @fabian-hiller | - | ✅ |
| notLength | @fabian-hiller | - | ✅ |
| notSize | @fabian-hiller | - | ✅ |
| notValue | @fabian-hiller | - | ✅ |
| octal | @ariskemper | #415 | ✅ |
| regex | @fabian-hiller | - | ✅ |
| safeInteger | @fabian-hiller | - | ✅ |
| size | @fabian-hiller | - | ✅ |
| some | @fabian-hiller | - | ✅ |
| startsWith | @fabian-hiller | - | ✅ |
| ulid | @fabian-hiller | - | ✅ |
| url | @fabian-hiller | - | ✅ |
| uuid | @fabian-hiller | - | ✅ |
| value | @fabian-hiller | - | ✅ |

</details>

<details>
  <summary>Transformations</summary><br />

| API | Author | Pull Request | Merged |
| -- | -- | -- | -- |
| toCustom | @fabian-hiller | | |
| toLowerCase | @fabian-hiller | | |
| toMaxValue | @fabian-hiller | | |
| toMinValue | @fabian-hiller | | |
| toTrimmed | @fabian-hiller | | |
| toTrimmedEnd | @fabian-hiller | | |
| toTrimmedStart | @fabian-hiller | | |
| toUpperCase | @fabian-hiller | | |

</details>

<details>
  <summary>Error</summary><br />

| API | Author | Pull Request | Merged |
| -- | -- | -- | -- |
| flatten | @fabian-hiller | | |
| ValiError | @fabian-hiller | | |

</details>

<details>
  <summary>Async</summary><br />

| API | Author | Pull Request | Merged |
| -- | -- | -- | -- |
| anyAsync | | | |
| arrayAsync | | | |
| bigintAsync | | | |
| blobAsync | | | |
| booleanAsync | | | |
| coerceAsync | | | |
| customAsync | | | |
| dateAsync | | | |
| enumAsync | | | |
| fallbackAsync | | | |
| forwardAsync | | | |
| getDefaultAsync | | | |
| getDefaultsAsync | | | |
| getFallbackAsync | | | |
| getFallbacksAsync | | | |
| instanceAsync | | | |
| intersectAsync | | | |
| literalAsync | | | |
| mapAsync | | | |
| mergeAsync | | | |
| nanAsync | | | |
| neverAsync | | | |
| nonNullableAsync | | | |
| nonNullishAsync | | | |
| nonOptionalAsync | | | |
| nullableAsync | | | |
| nullishAsync | | | |
| nullAsync | | | |
| numberAsync | | | |
| objectAsync | | | |
| omitAsync | | | |
| optionalAsync | | | |
| parseAsync | | | |
| partialAsync | | | |
| pickAsync | | | |
| picklistAsync | | | |
| recordAsync | | | |
| recursiveAsync | | | |
| requiredAsync | | | |
| safeParseAsync | | | |
| setAsync | | | |
| specialAsync | | | |
| stringAsync | | | |
| symbolAsync | | | |
| toCustomAsync | | | |
| transformAsync | | | |
| tupleAsync | | | |
| undefinedAsync | | | |
| unionAsync | | | |
| unknownAsync | | | |
| variantAsync | | | |
| voidAsync | | | |

**Comment by kazizi55** [comment]
@fabian-hiller 
Hi! To begin with, I'd like to take over `array` , `number` and `object`! 😄

**Comment by wout-junius** [comment]
@fabian-hiller I do not mind beginning on the `date`, `boolean`, and `symbol`! :D

**Comment by fabian-hiller** [comment]
Thank you guys! I updated the tables.

**Comment by ariskemper** [comment]
@fabian-hiller please update table for bic, mac, mac48, mac64, and include me.

**Comment by sqmasep** [comment]
@fabian-hiller Hi! I'll work on documenting `fallback`, `getFallback`, `getFallbacks`, `getDefault`, `getDefaults`, `optional`, `nullish`, & `nullable`, you can update the table 😉

**Comment by kazizi55** [comment]
@fabian-hiller 
Happy new year 2024! 😄 
I'd like to take over `is`, `keyof`, `merge`, and `omit`!

**Comment by fabian-hiller** [comment]
Happy new new year and thank you! I will update the tables.

**Comment by mwskwong** [comment]
Let me give `nonNullable` a try

**Comment by fabian-hiller** [comment]
I will take the rest of the API reference (except the async APIs) and finish them by the end of the month. You can still help by copying the reference of an already added sync APIs and adding its async reference. Just send here the async APIs you want to work on and I will add your username to the table.

**Comment by EltonLobo07** [comment]
@fabian-hiller Hi! Can I work on documenting `booleanAsync`, `stringAsync` & `numberAsync`? If yes, can you update the table?

**Comment by fabian-hiller** [comment]
Thank you for your help! We are currently rewriting the library in #502. Therefore, I recommend waiting until this PR is merged. Also, some APIs like `stringAsync` will not exist after the rewrite. See the PR description for more details.

**Comment by saku-1101** [comment]
Hi @fabian-hiller ! I also would love to work on documenting `minLength` and `maxLength`. Should I also wait until [#502](https://github.com/fabian-hiller/valibot/pull/502) has been merged?

**Comment by fabian-hiller** [comment]
Yes, please wait. I will give an update here in about a month. Thank you!

**Comment by fabian-hiller** [comment]
You can now go ahead if you like and add content to our async API references by copying the content from the sync version.

**Comment by EltonLobo07** [comment]
Hi @fabian-hiller! I would like to work on documenting `customAsync`, `arrayAsync` and `mapAsync`.

**Comment by fabian-hiller** [comment]
@EltonLobo07 I am very grateful for your contributions! Your work is always of great quality! 🙏

**Comment by EltonLobo07** [comment]
@fabian-hiller I would like to work on documenting `objectAsync`, `checkAsync` and `setAsync` next.

**Comment by fabian-hiller** [comment]
Yes, thank you!

**Comment by EltonLobo07** [comment]
@fabian-hiller I would like to work on documenting `awaitAsync`, `fallbackAsync`, `forwardAsync`, `getDefaultsAsync`, and `getFallbacksAsync` next.

**Comment by fabian-hiller** [comment]
Go ahead 🙏

**Comment by mehm8128** [comment]
@fabian-hiller Hi, I would like to work on documenting `tupleAsync`.

**Comment by fabian-hiller** [comment]
Yes, thank you!

**Comment by fabian-hiller** [comment]
@mehm8128 feel free to also work on `looseTupleAsync`, `tupleWithRestAsync` and `strictTupleAsync`.

**Comment by mehm8128** [comment]
I'll challenge it on another day🙌

**Comment by EltonLobo07** [comment]
@fabian-hiller I would like to work on documenting `intersectAsync`, `lazyAsync`, `looseObjectAsync`, `nonNullableAsync`, `nonNullishAsync`, `nonOptionalAsync`, and `strictObjectAsync` next.

**Comment by fabian-hiller** [comment]
Thank you!

**Comment by santoshyadavdev** [comment]
Hey @fabian-hiller ,

I will take Transformations APIs.

**Comment by EltonLobo07** [comment]
@fabian-hiller, I would like to work on documenting `parseAsync`, `parserAsync`, `unionAsync` and `variantAsync` next.

**Comment by fabian-hiller** [comment]
Thank you both!

**Comment by fabian-hiller** [comment]
@santoshyadavdev any updates?

**Comment by EltonLobo07** [comment]
@fabian-hiller I would like to work on documenting `partialCheckAsync`, `pipeAsync`, `safeParseAsync` and `safeParserAsync` next.

**Comment by fabian-hiller** [comment]
Thank you @EltonLobo07!

**Comment by santoshyadavdev** [comment]
@fabian-hiller I see some of them are already documented under actions

**Comment by fabian-hiller** [comment]
Yes, unfortunately I stopped updating the table due to time constraints after finishing all the sync API references. Now only a few async references are missing.

**Comment by santoshyadavdev** [comment]
no worries i will open another PR

**Comment by santoshyadavdev** [comment]
@fabian-hiller I tried to serve the app locally and getting this error, do you know how to fix it

<img width="1451" alt="Screenshot 2024-09-01 at 15 39 39" src="https://github.com/user-attachments/assets/494f9d1b-a7dd-4f16-b2de-76f0211cf33d">

**Comment by fabian-hiller** [comment]
Yes, I think you need to build the library first: `cd ./library && pnpm build`

**Comment by EltonLobo07** [comment]
@fabian-hiller I would like to work on documenting `partialAsync`, `rawCheckAsync`, `recordAsync`, and `requiredAsync`. @santoshyadavdev Can you let us know which APIs you are planning on documenting? So that we can eliminate the possibility of documenting the same APIs. Also, here is the updated missing list: `nullableAsync`, `nullishAsync`, `optionalAsync`, `rawTransformAsync`, `transformAsync`, and `undefinedableAsync`.

**Comment by santoshyadavdev** [comment]
@EltonLobo07 I started https://github.com/fabian-hiller/valibot/pull/810 documenting 

nullableAsync, nullishAsync go ahead wit partialAsync, rawCheckAsync, recordAsync, and requiredAsync I will remove them from my PR

**Comment by fabian-hiller** [comment]
We did it! Thanks to everyone who helped, and a special thanks to @EltonLobo07 for adding almost all of the async documentation.
